internal/cluster/certificates.go

```diff
@@ -67,13 +74,65 @@ var (
 // Certificates are the certificates necessary to bootstrap a cluster.
 type Certificates []*Certificate
 
-// NewCertificates return an initialized but empty set of CA certificates needed to bootstrap a cluster.
-func NewCertificates() Certificates {
+// NewCertificatesForControlPlane returns a list of certificates configured for a control plane node
+func NewCertificatesForControlPlane(config *v1beta1.ClusterConfiguration) Certificates {
+	if config.CertificatesDir == "" {
+		config.CertificatesDir = defaultCertificatesDir
+	}
+
+	certificates := Certificates{
+		&Certificate{
+			Purpose:  secret.ClusterCA,
+			CertFile: filepath.Join(config.CertificatesDir, "ca.crt"),
+			KeyFile:  filepath.Join(config.CertificatesDir, "ca.key"),
+		},
+		&Certificate{
+			Purpose:  ServiceAccount,
+			CertFile: filepath.Join(config.CertificatesDir, "sa.pub"),
+			KeyFile:  filepath.Join(config.CertificatesDir, "sa.key"),
+		},
+		&Certificate{
+			Purpose:  FrontProxyCA,
+			CertFile: filepath.Join(config.CertificatesDir, "front-proxy-ca.crt"),
+			KeyFile:  filepath.Join(config.CertificatesDir, "front-proxy-ca.key"),
+		},
+	}
+
+	etcdCert := &Certificate{
+		Purpose:  EtcdCA,
+		CertFile: filepath.Join(config.CertificatesDir, "etcd", "ca.crt"),
+		KeyFile:  filepath.Join(config.CertificatesDir, "etcd", "ca.key"),
+	}
+
+	// TODO make sure all the fields are actually defined and return an error if not
+	if config.Etcd.External != nil {
+		etcdCert = &Certificate{
+			Purpose:  EtcdCA,
+			CertFile: config.Etcd.External.CAFile,
+		}
+		apiserverEtcdClientCert := &Certificate{
+			Purpose:  APIServerEtcdClient,
+			CertFile: config.Etcd.External.CertFile,
+			KeyFile:  config.Etcd.External.KeyFile,
+		}
+		certificates = append(certificates, apiserverEtcdClientCert)
+	}
+
+	certificates = append(certificates, etcdCert)
+	return certificates
+}
+
+// NewCertificatesForWorker return an initialized but empty set of CA certificates needed to bootstrap a cluster.
+func NewCertificatesForWorker(caCertPath string) Certificates {
+	if caCertPath == "" {
+		caCertPath = defaultCertificatesDir + "ca.crt"
```

Comment:
Thanks @chuckha :+1: 
But I prefer to write like this 
```suggestion
		caCertPath = filepath.Join(defaultCertificatesDir, "ca.crt")
```
WDYT ?

Reply:
Yes, this is objectively better